src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -256,18 +256,48 @@ public function denormalize($data, $class, $format = null, array $context = arra
                 continue;
             }
 
-            $value = $this->validateAndDenormalize($class, $attribute, $value, $format, $context);
+            try {
+                $value = $this->validateAndDenormalize($class, $attribute, $value, $format, $context, $accumulatingContext);
+            } catch (NotDenormalizableValueException $exception) {
+                if (isset($context[self::COLLECT_ALL_ERRORS]) && $context[self::COLLECT_ALL_ERRORS]) {
+                    $accumulatingContext[$attribute][] = $exception->getMessage();
+
+                    continue;
+                }
+
+                throw $exception;
+            } catch (UnexpectedValuesException $exception) {
+                $accumulatingContext[$attribute] = $exception->getUnexpectedValueErrors();
+                continue;
+            }
+
+            if (null === $value) {
+                continue;
+            }
+
             try {
                 $this->setAttributeValue($object, $attribute, $value, $format, $context);
             } catch (InvalidArgumentException $e) {
-                throw new NotNormalizableValueException($e->getMessage(), $e->getCode(), $e);
+                $exception = new NotDenormalizableValueException($e->getMessage(), $attribute, $e);
+
+                if (isset($context[self::COLLECT_ALL_ERRORS]) && $context[self::COLLECT_ALL_ERRORS]) {
+                    $accumulatingContext[$attribute][] = $exception->getMessage();
+
+                    continue;
+                }
+
+                throw $exception;
             }
         }
 
         if (!empty($extraAttributes)) {
             throw new ExtraAttributesException($extraAttributes);
```

Comment:
Shouldn't this exception be collected too? It will be useful when creating web APIs to be able to return all errors in one time to the client, including this one.